File: webapp/src/Websocket/types/WebsocketMessage.ts
import {RpcRequest, RpcResponse} from "./Rpc";
import {Notify} from "./Subscription";

export type WebsocketMessage = {
    id: string;
    type: WebsocketMessageType;
    rpcRequest?: RpcRequest;
    rpcResponse?: RpcResponse;
    notify?: Notify;
}

export type WebsocketMessageType = 'rpcRequest' | 'rpcResponse' | 'notify';

File: src/main/kotlin/com/dehnes/smarthome/api/dtos/WebsocketMessage.kt
package com.dehnes.smarthome.api.dtos

data class WebsocketMessage(
    val id: String,
    val type: WebsocketMessageType,
    val rpcRequest: RpcRequest? = null,
    val rpcResponse: RpcResponse? = null,
    val notify: Notify? = null
)

enum class WebsocketMessageType {
    rpcRequest,
    rpcResponse,
    notify
}


